### PR TITLE
Update TMC endpoint mapping rules for Tanzu context

### PR DESCRIPTION
**Pull request by anujc25**
### What this PR does / why we need it
- Update TMC endpoint mapping rules for Tanzu context
### Which issue(s) this PR fixes

Fixes #

### Describe testing done for PR
- Unit tests

### Release note

```release-note
Update TMC endpoint mapping rules for the Tanzu context
```

### CODE REVIEW

File: pkg/command/context.go [hunk 1]
```diff
@@ -2143,14 +2142,30 @@ func renderDynamicTable(slices interface{}, tableWriter component.OutputWriter,
...
 func mapTanzuEndpointToTMCEndpoint(tanzuEndpoint string) string {
-	// Define the regular expression pattern
-	apiPattern := regexp.MustCompile(`https://api\.tanzu(-\w*)?\.cloud\.vmware\.com`)
-	// Replace "api" with "tmc" in the input URL
-	tmcEndpoint := apiPattern.ReplaceAllString(tanzuEndpoint, "https://tmc.tanzu$1.cloud.vmware.com")
+	tmcEndpoint := ""
+	// Define the mapping rules
+	mappingRules := []struct {
+		prefix string
+		suffix string
```
Comment (vuil) [inline pkg/command/context.go:2149]: nit: it is technically not a suffix but an ops specific prefix.

File: pkg/command/context.go [hunk 2]
```diff
@@ -2143,14 +2142,30 @@ func renderDynamicTable(slices interface{}, tableWriter component.OutputWriter,
...
+	}{
+		{"https://api.tanzu", "https://tmc.tanzu"},
```
Comment (vuil) [inline pkg/command/context.go:2151]: wondering: should we be concerned that we are replacing for not just .cloud.vmware.com/.tanzu.broad.com but _every_ other domain?

Comment (anujc25) [reply pkg/command/context.go:2151]: Yeah. I wanted to keep things flexible as much as possible for now. So this is intentional. Do you think we need specific one?

File: pkg/command/context.go [hunk 3]
```diff
@@ -2143,14 +2147,35 @@ func renderDynamicTable(slices interface{}, tableWriter component.OutputWriter,
...
+		tmcEndpointPrefix   string
+	}{
+		{"https://api.tanzu", "https://tmc.tanzu"},
+		{"https://ucp.platform", "https://ops.platform"},
+	}
+
+	// Iterate through the mapping rules
+	for _, rule := range mappingRules {
+		if strings.HasPrefix(tanzuEndpoint, rule.tanzuEndpointPrefix) {
+			// Replace the prefix with the suffix
```
Comment (vuil) [inline pkg/command/context.go:2168]: nit: comment needs update

**Review by vuil — APPROVED** [review]
nit about a comment update, but lgtm, thanks.

File: pkg/command/context.go (hunk 2 above)
Comment (anujc25) [reply pkg/command/context.go:2161]: We had an offline discussion and decided to keep this flexible as we have in this change.

**Review by vuil — APPROVED** [review]

**Pull request merged**